pkg/network/setup/filters_test.go

```diff
@@ -47,4 +49,141 @@ var _ = Describe("Network setup filters", func() {
 		})
 	})
+
+	Context("FilterNetsForLiveNICChange", func() {
+		const (
+			net1Name = "net1"
+			net2Name = "net2"
+			nad1Name = "nad1"
+			nad2Name = "nad2"
+		)
+
+		multusAndDomainInfoSource := vmispec.NewInfoSource(vmispec.InfoSourceMultusStatus, vmispec.InfoSourceDomain)
+
+		It("Should return an empty list when there are no networks to hot plug/unplug", func() {
+			vmi := libvmi.New(
+				libvmi.WithInterface(*v1.DefaultBridgeNetworkInterface()),
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding(net1Name)),
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding(net2Name)),
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+				libvmi.WithNetwork(libvmi.MultusNetwork(net1Name, nad1Name)),
+				libvmi.WithNetwork(libvmi.MultusNetwork(net2Name, nad2Name)),
+				libvmistatus.WithStatus(
+					libvmistatus.New(
+						libvmistatus.WithInterfaceStatus(v1.VirtualMachineInstanceNetworkInterface{
+							Name:       "default",
+							InfoSource: vmispec.InfoSourceDomainAndGA,
+						}),
+						libvmistatus.WithInterfaceStatus(v1.VirtualMachineInstanceNetworkInterface{
+							Name:       net1Name,
+							InfoSource: multusAndDomainInfoSource,
+						}),
+						libvmistatus.WithInterfaceStatus(v1.VirtualMachineInstanceNetworkInterface{
+							Name:       net2Name,
+							InfoSource: multusAndDomainInfoSource,
+						}),
+					),
+				),
+			)
+
+			Expect(network.FilterNetsForLiveNICChange(vmi)).To(BeEmpty())
+		})
+
+		It("Should return a network to hotplug", func() {
+			multusAndDomainInfoSource := vmispec.NewInfoSource(vmispec.InfoSourceMultusStatus, vmispec.InfoSourceDomain)
+			vmi := libvmi.New(
+				libvmi.WithInterface(*v1.DefaultBridgeNetworkInterface()),
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding(net1Name)),
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding(net2Name)),
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+				libvmi.WithNetwork(libvmi.MultusNetwork(net1Name, nad1Name)),
+				libvmi.WithNetwork(libvmi.MultusNetwork(net2Name, nad2Name)),
+				libvmistatus.WithStatus(
+					libvmistatus.New(
+						libvmistatus.WithInterfaceStatus(v1.VirtualMachineInstanceNetworkInterface{
+							Name:       "default",
+							InfoSource: vmispec.InfoSourceDomainAndGA,
+						}),
+						libvmistatus.WithInterfaceStatus(v1.VirtualMachineInstanceNetworkInterface{
+							Name:       net1Name,
+							InfoSource: multusAndDomainInfoSource,
+						}),
+						libvmistatus.WithInterfaceStatus(v1.VirtualMachineInstanceNetworkInterface{
+							Name:       net2Name,
+							InfoSource: vmispec.InfoSourceMultusStatus,
+						}),
+					),
+				),
+			)
+
+			Expect(network.FilterNetsForLiveNICChange(vmi)).To(Equal([]v1.Network{*libvmi.MultusNetwork(net2Name, nad2Name)}))
+		})
+
+		It("Should return a network to hotunplug", func() {
+			absentIface := libvmi.InterfaceDeviceWithBridgeBinding(net1Name)
```

Comment:
nit:
Having the absent interface definition outside the VMI definition is cumbersome,
and makes me go back to the interface definition, check the name, go back to the VMI defintion and check there is a matching network name.

Please consider inlining this interface definition in the VMI definition.
It may require introducing new libvmi option, I think it can be defined locally for this file only.
Or define another variable for the associated network (e.g.: absentNetwork)

Reply:
IIRC, @nirdothan has added such a function in the PR to introduce link state.
We can do it in a follow-up after his PR will be merged.

Reply:
I think it worth having it in this PR as well, it would make reviewing those tests easier.